From de5bd01aecfe4bf9b399b71e92dc4c05a83266a4 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.L-H@gmx.de>
Date: Fri, 17 May 2013 04:58:00 +0200
Subject: [PATCH 162/390] rtl8192du: drop unused private headers

 - the private include/wireless.h isn't included anywhere
 - while include/circ_buf.h is included several times, none of its
   definitions are actually used.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 core/rtw_recv.c         |  1 -
 core/rtw_xmit.c         |  1 -
 hal/rtl8192d_cmd.c      |  1 -
 hal/rtl8192du_recv.c    |  1 -
 hal/rtl8192du_xmit.c    |  1 -
 hal/usb_ops_linux.c     |  1 -
 include/circ_buf.h      | 27 ----------------
 include/osdep_service.h |  1 -
 include/wireless.h      | 83 -------------------------------------------------
 os_dep/xmit_linux.c     |  1 -
 10 files changed, 118 deletions(-)
 delete mode 100644 include/circ_buf.h
 delete mode 100644 include/wireless.h

--- a/drivers/staging/rtl8192du/core/rtw_recv.c
+++ b/drivers/staging/rtl8192du/core/rtw_recv.c
@@ -28,7 +28,6 @@
 #include <ethernet.h>
 #include <usb_ops.h>
 #include <wifi.h>
-#include <circ_buf.h>
 
 #ifdef CONFIG_NEW_SIGNAL_STAT_PROCESS
 void rtw_signal_stat_timer_hdl(RTW_TIMER_HDL_ARGS);
--- a/drivers/staging/rtl8192du/core/rtw_xmit.c
+++ b/drivers/staging/rtl8192du/core/rtw_xmit.c
@@ -24,7 +24,6 @@
 #include <drv_types.h>
 #include <wifi.h>
 #include <osdep_intf.h>
-#include <circ_buf.h>
 #include <linux/ip.h>
 
 #include <usb_ops.h>
--- a/drivers/staging/rtl8192du/hal/rtl8192d_cmd.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_cmd.c
@@ -25,7 +25,6 @@
 #include <recv_osdep.h>
 #include <cmd_osdep.h>
 #include <mlme_osdep.h>
-#include <circ_buf.h>
 #include <rtw_ioctl_set.h>
 
 #include <rtl8192d_hal.h>
--- a/drivers/staging/rtl8192du/hal/rtl8192du_recv.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192du_recv.c
@@ -28,7 +28,6 @@
 #include <ethernet.h>
 #include <usb_ops.h>
 #include <wifi.h>
-#include <circ_buf.h>
 #include <rtl8192d_hal.h>
 
 void rtl8192du_init_recvbuf(struct rtw_adapter *padapter, struct recv_buf *precvbuf)
--- a/drivers/staging/rtl8192du/hal/rtl8192du_xmit.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192du_xmit.c
@@ -23,7 +23,6 @@
 #include <drv_types.h>
 #include <wifi.h>
 #include <osdep_intf.h>
-#include <circ_buf.h>
 #include <usb_ops.h>
 #include <rtl8192d_hal.h>
 
--- a/drivers/staging/rtl8192du/hal/usb_ops_linux.c
+++ b/drivers/staging/rtl8192du/hal/usb_ops_linux.c
@@ -24,7 +24,6 @@
 #include <drv_types.h>
 #include <osdep_intf.h>
 #include <usb_ops.h>
-#include <circ_buf.h>
 #include <recv_osdep.h>
 #include <rtl8192d_hal.h>
 
--- a/drivers/staging/rtl8192du/include/circ_buf.h
+++ /dev/null
@@ -1,27 +0,0 @@
-/******************************************************************************
- *
- * Copyright(c) 2007 - 2011 Realtek Corporation. All rights reserved.
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of version 2 of the GNU General Public License as
- * published by the Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
- *
- *
- ******************************************************************************/
-#ifndef __CIRC_BUF_H_
-#define __CIRC_BUF_H_ 1
-
-#define CIRC_CNT(head,tail,size) (((head) - (tail)) & ((size)-1))
-
-#define CIRC_SPACE(head,tail,size) CIRC_CNT((tail),((head)+1),(size))
-
-#endif /* _CIRC_BUF_H_ */
--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -40,7 +40,6 @@
 #endif
 	#include <linux/netdevice.h>
 	#include <linux/skbuff.h>
-	#include <linux/circ_buf.h>
 	#include <asm/uaccess.h>
 	#include <asm/byteorder.h>
 	#include <asm/atomic.h>
--- a/drivers/staging/rtl8192du/include/wireless.h
+++ /dev/null
@@ -1,83 +0,0 @@
-/******************************************************************************
- *
- * Copyright(c) 2007 - 2011 Realtek Corporation. All rights reserved.
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of version 2 of the GNU General Public License as
- * published by the Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
- * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
- *
- *
- ******************************************************************************/
-
-#ifndef _LINUX_WIRELESS_H
-#define _LINUX_WIRELESS_H
-
-/***************************** INCLUDES *****************************/
-
-#define __user
-#include <sys/socket.h>			/* for "struct sockaddr" et al	*/
-#include <net/if.h>			/* for IFNAMSIZ and co... */
-
-/****************************** TYPES ******************************/
-
-/* --------------------------- SUBTYPES --------------------------- */
-/*
- *	For all data larger than 16 octets, we need to use a
- *	pointer to memory allocated in user space.
- */
-struct	iw_point
-{
-  void __user	*pointer;	/* Pointer to the data  (in user space) */
-  __u16		length;		/* number of fields or size in bytes */
-  __u16		flags;		/* Optional params */
-};
-
-
-/* ------------------------ IOCTL REQUEST ------------------------ */
-/*
- * This structure defines the payload of an ioctl, and is used
- * below.
- *
- * Note that this structure should fit on the memory footprint
- * of iwreq (which is the same as ifreq), which mean a max size of
- * 16 octets = 128 bits. Warning, pointers might be 64 bits wide...
- * You should check this when increasing the structures defined
- * above in this file...
- */
-union	iwreq_data
-{
-	/* Config - generic */
-	char		name[IFNAMSIZ];
-	/* Name : used to verify the presence of  wireless extensions.
-	 * Name of the protocol/provider... */
-
-	struct iw_point	data;		/* Other large parameters */
-};
-
-/*
- * The structure to exchange data for ioctl.
- * This structure is the same as 'struct ifreq', but (re)defined for
- * convenience...
- * Do I need to remind you about structure size (32 octets) ?
- */
-struct	iwreq
-{
-	union
-	{
-		char	ifrn_name[IFNAMSIZ];	/* if name, e.g. "eth0" */
-	} ifr_ifrn;
-
-	/* Data part (defined just above) */
-	union	iwreq_data	u;
-};
-
-#endif	/* _LINUX_WIRELESS_H */
--- a/drivers/staging/rtl8192du/os_dep/xmit_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/xmit_linux.c
@@ -29,7 +29,6 @@
 #include <mlme_osdep.h>
 #include <xmit_osdep.h>
 #include <osdep_intf.h>
-#include <circ_buf.h>
 
 
 uint rtw_remainder_len(struct pkt_file *pfile)
